File: programmers/etc/level2/귤_고르기.py
from collections import Counter

def solution(k, tangerine):
    answer = 0
    cnt = sorted(Counter(tangerine).values(), reverse=True)
    
    for c in cnt:
        k -= c
        answer += 1
        if k <= 0:
            break
    
    return answer


k = 6
tangerine = [1, 3, 2, 5, 4, 5, 2, 3]
print(solution(k, tangerine)) # 3

k = 4
tangerine = [1, 3, 2, 5, 4, 5, 2, 3]
print(solution(k, tangerine)) # 2

k = 2
tangerine = [1, 1, 1, 1, 2, 2, 2, 3]
print(solution(k, tangerine)) # 1

File: Java/programmers/etc/level2/귤고르기.java
package CodeTest.Java.programmers.etc.level2;

import java.util.*;

public class 귤고르기 {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        HashMap<Integer, Integer> map = new HashMap<>();
        
        for (int t : tangerine) {
            map.put(t, map.getOrDefault(t, 0)+1);
        }
        
        List<Integer> cnt = new ArrayList<>(map.keySet());
        cnt.sort((o1,o2) -> {return map.get(o2)-map.get(o1);});
        
        for (int c : cnt) {
            k -= map.get(c);
            answer++;
            if (k <= 0) break;
        }
        
        return answer;
    }
}
